src/collector.js

```diff
@@ -0,0 +1,53 @@
+import { default as Message } from 'message';
+
+// "I have a display case ready and waiting for our newest acquistions!"
+// --Taneleer Tivan
+
+
+export default class Collector {
+
+  constructor() {
+    this.errors = [];
+    this.notices = [];
+    this.warnings = [];
+  }
+
+  getMessageList(type) {
+    switch (type) {
+      case 'error':
+        return this.errors;
+      case 'notice':
+        return this.notices;
+      case 'warning':
+        return this.warnings;
+      default:
+        throw new Error(`Invalid message type "${type}"`);
+    }
+  }
+
+  messageCount() {
+    return this.errors.length +
+      this.notices.length +
+      this.warnings.length;
+  }
+
+  _addMessage(type, opts) {
+    // Message will throw for incorrect types.
+    // we have a test to ensure that is the case.
+    var message = new Message(type, opts);
+    var list = this.getMessageList(type);
+    list.push(message);
+  }
+
+  addError(opts) {
```

Comment:
Having to supply an empty object as an argument seems silly. We should make the `opts={}` default to an empty object if that's all it needs, or make a note to make it validate the presence of certain properties in the future.

Lower in the tests there's `addError({})`, which seems awkward.